### PR TITLE
EOS-20688: Start and Stop MessageBus rest service in utils_setup

### PR DESCRIPTION
**Pull request by selvakumaar5496**
# Problem Statement
- Start and Stop MessageBus rest service in utils_setup

# Design
- start MessageBus rest service in utils_setup init phase
- stop MessageBus rest service in utils_setup reset phase

# Coding 
-  Coding conventions are followed and code is consistent [Y/N]: Y
-  Confirm All CODACY errors are resolved [Y/N]: Y

# Testing 
- [ ] Confirm that Test Cases are added (for both the cases, fix and feature) [Y/N]: N 
- [ ] Confirm Test Cases cover Happy Path, Non-Happy Path and Scalability [Y/N]: N
- [ ] Confirm Testing was performed with installed RPM [Y/N]: N
Reason: Utils_setup do not have test cases. It is tested by doing provisioning in a fresh local vm

# Review Checklist 
  Before posting the PR please ensure
- [x] PR is self reviewed
- [x] Jira is updated
- [x] Check if the description is clear and explained. 
- [x] Check Acceptance Criterion is defined. 
- [ ] All the tests performed should be mentioned before Resolving a JIRA. 
- [ ] Verification needs to be done before marked as Closed/Verified 

# Documentation
- [ ] Changes done to WIKI / Confluence page

### CODE REVIEW

File: py-utils/src/setup/utils.py [hunk 1]
```diff
@@ -163,6 +163,17 @@ def init():
             admin.register_message_type(message_types=['IEM'], partitions=1)
         except MessageBusError as e:
             raise SetupError(e.rc, "Unable to create message_type. %s", e)
+
+        # Start MessageBus rest service
+        try:
+            rc = os.system("service cortx_message_bus start")
+        except OSError as e:
```
Comment (rahul-tripathi-git) [inline py-utils/src/setup/utils.py:170]: If service is not available, there will not be any OSError, Hence in this case we might need to use simpleprocess.

File: py-utils/src/setup/utils.py [hunk 2]
```diff
@@ -216,6 +227,17 @@ def reset():
             except Exception as e:
                 raise SetupError(errors.OP_FAILED, "Can not reset Message Bus. \
                     %s", e)
+
+        # Stop MessageBus rest service
+        try:
+            rc = os.system("service cortx_message_bus stop")
+        except OSError as e:
```
Comment (rahul-tripathi-git) [inline py-utils/src/setup/utils.py:234]: Same applies here, we might need to check return code.

**Review by rahul-tripathi-git — CHANGES_REQUESTED** [review]

File: py-utils/src/setup/utils.py (hunk 1 above)
Comment (selvakumaar5496) [reply py-utils/src/setup/utils.py:170]: again checking rc value. Please check below

File: py-utils/src/setup/utils.py [hunk 3]
```diff
@@ -163,6 +163,17 @@ def init():
...
         except MessageBusError as e:
             raise SetupError(e.rc, "Unable to create message_type. %s", e)
+
+        # Start MessageBus rest service
+        try:
+            rc = os.system("service cortx_message_bus start")
+        except OSError as e:
+            raise SetupError(e.errno, "Unable to start MessageBus rest service. \
+                %s", e)
+        if rc != 0:
```
Comment (ujjwalpl) [inline py-utils/src/setup/utils.py:173]: os.system('command') returns a 16 bit number, which first 8 bits from left(lsb) talks about signal used by os to close the command, Next 8 bits talks about return code of command. rc cant be compared with 0. You need to obtain high order 8 bits, from rc. 

```
>>> rc = os.system("ls /dfsfds")
ls: cannot access /dfsfds: No such file or directory
>>> print(rc)
512
>>> print(rc>>8)
2
```

File: py-utils/src/setup/utils.py [hunk 4]
```diff
@@ -216,6 +227,17 @@ def reset():
...
                 raise SetupError(errors.OP_FAILED, "Can not reset Message Bus. \
                     %s", e)
+
+        # Stop MessageBus rest service
+        try:
+            rc = os.system("service cortx_message_bus stop")
+        except OSError as e:
+            raise SetupError(e.errno, "Unable to stop MessageBus rest service. \
+                %s", e)
+        if rc != 0:
```
Comment (ujjwalpl) [inline py-utils/src/setup/utils.py:237]: Same here

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

File: py-utils/src/setup/utils.py (hunk 3 above)
Comment (selvakumaar5496) [reply py-utils/src/setup/utils.py:173]: done

File: py-utils/src/setup/utils.py (hunk 4 above)
Comment (selvakumaar5496) [reply py-utils/src/setup/utils.py:237]: Done

File: py-utils/src/setup/utils.py (hunk 2 above)
Comment (selvakumaar5496) [reply py-utils/src/setup/utils.py:234]: again checking rc value. Please check below

File: py-utils/src/setup/utils.py [hunk 5]
```diff
@@ -163,6 +163,17 @@ def init():
             admin.register_message_type(message_types=['IEM'], partitions=1)
         except MessageBusError as e:
             raise SetupError(e.rc, "Unable to create message_type. %s", e)
+
+        # Start MessageBus rest service
+        try:
```
Comment (ujjwalpl) [inline py-utils/src/setup/utils.py:168]: systemctl requires root privileges. Please add a check somewehere in the validate that root privilege is present.

File: py-utils/src/setup/utils.py (hunk 1 above)
Comment (ujjwalpl) [reply py-utils/src/setup/utils.py:170]: os.system is incorrect logic. It does not throw exception. 

```
>>> try:
...     rc = os.system("systemctl start xxx")
... except:
...     print("exception")
...
Failed to start xxx.service: Unit not found.
>>>
```

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

File: py-utils/src/setup/utils.py (hunk 5 above)
Comment (selvakumaar5496) [reply py-utils/src/setup/utils.py:168]: running as root

File: py-utils/src/setup/utils.py (hunk 1 above)
Comment (selvakumaar5496) [reply py-utils/src/setup/utils.py]: used SimpleProcess

File: py-utils/src/setup/utils.py (hunk 5 above)
Comment (ujjwalpl) [reply py-utils/src/setup/utils.py:168]: We need to add a check in case someone runs it as non root user.

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

File: py-utils/src/setup/utils.py [hunk 6]
```diff
@@ -163,6 +163,20 @@ def init():
...
+            cmd = "systemctl start cortx_message_bus"
+            cmd_proc = SimpleProcess(cmd)
+            _, res_err, res_rc = cmd_proc.run()
+
+            if res_rc != 0:
+                raise SetupError(res_rc, "Unable to start MessageBus rest \
```
Comment (ujjwalpl) [inline py-utils/src/setup/utils.py:174]: "Unable to start MessageBus Service. %s"

File: py-utils/src/setup/utils.py [hunk 7]
```diff
@@ -163,6 +163,20 @@ def init():
...
+                    service. %s", res_err.decode('utf-8'))
+        except Exception as e:
```
Comment (ujjwalpl) [inline py-utils/src/setup/utils.py:176]: No need to handle exception. SimpleProcess already does this.

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

File: py-utils/src/setup/utils.py [hunk 8]
```diff
@@ -216,6 +230,20 @@ def reset():
...
+        # Stop MessageBus rest service
+        try:
+            cmd = "systemctl stop cortx_message_bus"
+            cmd_proc = SimpleProcess(cmd)
+            _, res_err, res_rc = cmd_proc.run()
+
+            if res_rc != 0:
+                raise SetupError(res_rc, "Unable to stop MessageBus rest \
+                    service. %s", res_err.decode('utf-8'))
+        except Exception as e:
```
Comment (ujjwalpl) [inline py-utils/src/setup/utils.py:243]: No need to handle exception. SimpleProcess does not throw any exception.

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

File: py-utils/src/setup/utils.py (hunk 8 above)
Comment (selvakumaar5496) [reply py-utils/src/setup/utils.py]: done, removed

File: py-utils/src/setup/utils.py (hunk 7 above)
Comment (selvakumaar5496) [reply py-utils/src/setup/utils.py]: done, removed

File: py-utils/src/setup/utils.py (hunk 6 above)
Comment (selvakumaar5496) [reply py-utils/src/setup/utils.py]: done

**Review by ujjwalpl — APPROVED** [review]

File: py-utils/src/setup/utils.py [hunk 9]
```diff
@@ -163,6 +163,15 @@ def init():
...
             raise SetupError(e.rc, "Unable to create message_type. %s", e)
+
+        # Start MessageBus Service
+        cmd = SimpleProcess("systemctl start cortx_message_bus")
+        _, stderr, res_rc = cmd.run()
+
+        if res_rc != 0:
+            raise SetupError(res_rc, "Unable to start MessageBus Service \
+                %s", stderr.decode('utf-8'))
+
```
Comment (sachinpunadikar) [inline py-utils/src/setup/utils.py:174]: After starting the service check the status to ensure it has been started correctly. And then return success or failure

**Review by sachinpunadikar — CHANGES_REQUESTED** [review]
Add status check after starting the service

File: py-utils/src/setup/utils.py (hunk 9 above)
Comment (selvakumaar5496) [reply py-utils/src/setup/utils.py:177]: Done

File: py-utils/src/setup/utils.py [hunk 10]
```diff
@@ -163,6 +163,22 @@ def init():
...
+            raise SetupError(start_rc, "Unable to start MessageBus Service \
+                %s", start_err.decode('utf-8'))
+
+        status_cmd = SimpleProcess("systemctl status cortx_message_bus")
+        _, status_err, status_rc = status_cmd.run()
+
+        if status_rc != 0:
+            raise SetupError(status_rc, "MessageBus Service is either failed \
+                inactive. %s", status_err.decode('utf-8'))
+
```
Comment (sachinpunadikar) [inline py-utils/src/setup/utils.py:181]: fix the codacy issue

**Review by sachinpunadikar — CHANGES_REQUESTED** [review]
fix codacy issue

**Review by sachinpunadikar — APPROVED** [review]
+1

**Review by rahul-tripathi-git — APPROVED** [review]
+1

**Comment by sachinpunadikar** [comment]
test this please

**Comment by sachinpunadikar** [comment]
test this please

**Comment by sachinpunadikar** [comment]
Custom VM deployment is going through clean - http://eos-jenkins.colo.seagate.com/job/GitHub-custom-ci-builds/job/centos-7.8/job/cortx-custom-ci/1528/

**Pull request merged**